mkosi/addon.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import argparse
+import os
+import sys
+import tempfile
+from pathlib import Path
+
+import mkosi.resources
+from mkosi.config import DocFormat, OutputFormat
+from mkosi.documentation import show_docs
+from mkosi.initrd import initrd_common_args, initrd_finalize, process_crypttab
+from mkosi.log import log_setup
+from mkosi.run import run, uncaught_exception_handler
+from mkosi.types import PathString
+from mkosi.util import resource_path
+
+
+@uncaught_exception_handler()
+def main() -> None:
+    log_setup()
+
+    parser = argparse.ArgumentParser(
+        prog="mkosi-addon",
+        description="Build initrd/cmdline/ucode addon for the current system using mkosi",
+        allow_abbrev=False,
+        usage="mkosi-addon [options...]",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="NAME",
+        help="Output name",
+        default="mkosi-local.addon.efi",
+    )
+
+    initrd_common_args(parser)
+
+    args = parser.parse_args()
+
+    if args.show_documentation:
+        with resource_path(mkosi.resources) as r:
+            show_docs("mkosi-addon", DocFormat.all(), resources=r)
+        return
+
+    # No local configuration? Then nothing to do
+    if not Path("/etc/mkosi-addon").exists() and not Path("/run/mkosi-addon").exists():
```

Comment:
Actually let's move this to the kernel-install plugin, makes more sense to do the skip there rather than here, having the skip here makes local development and testing harder than necessary